src/canmatrix/canmatrix.py

```diff
@@ -815,67 +815,55 @@ def calcDLC(self):
                 maxBit = sig.getStartbit() + int(sig.size)
         self.size = max(self.size, int(math.ceil(maxBit / 8)))
 
-    def findNotUsedBits(self):
+    def get_frame_layout(self):
         """
-        Find unused bits in frame.
+        get layout of frame.
 
-        Represents the bit usage in the frame by means of a list with 64 items.
-        Every item represents one bit and contains unique number for each signal, occupying that bit.
-        Numbering starts from one.
+        Represents the bit usage in the frame by means of a list with n items (n bits of frame length).
+        Every item represents one bit and contains a list of signals (object refs) with each signal, occupying that bit.
+        Bits with empty list are unused.
 
-        Bits with "zero" index are unused.
-
-        Example: [2, 2, 2, 1, 1, 0, 0, 3, 3, 3, 3, 0, 0, ...]
-
-        :return: list with signal "index plus one" in every bit. Zeros mean 'unused'.
-        :rtype: list of int
+        Example: [[], [], [], [sig1], [sig1], [sig1, sig5], [sig2, sig5], [sig2], []]
+        :return: list of lists with signalnames
+        :rtype: list of lists
         """
-        bitfield = []
-        bitfieldLe = []
-        bitfieldBe = []
-
-        for i in range(0,64):
-            bitfieldBe.append(0)
-            bitfieldLe.append(0)
-            bitfield.append(0)
-        i = 0
-
-        for sig in self.signals:
-            i += 1
-            for bit in range(sig.getStartbit(),  sig.getStartbit() + int(sig.size)):
-                if sig.is_little_endian:
-                    bitfieldLe[bit] = i
-                else:
-                    bitfieldBe[bit] = i
-
-        for i in range(0,8):
-            for j in range(0,8):
-                bitfield[i*8+j] = bitfieldLe[i*8+(7-j)]
+        little_bits = [[] for _dummy in range(0, (self.size * 8))]
+        big_bits = [[] for _dummy in range(0, (self.size * 8))]
+        for signal in self.signals:
+            if signal.is_little_endian:
+                least = self.size * 8 - signal.startBit
+                most = least - signal.size
+                for i in range(most, least):
+                    little_bits[i].append(signal)
+            else:
+                most = signal.startBit
+                least = most + signal.size
+                for i in range(most, least):
+                    little_bits[i].append(signal)
 
-        for i in range(0,8):
-            for j in range(0,8):
-                if bitfield[i*8+j] == 0:
-                    bitfield[i*8+j] = bitfieldBe[i*8+j]
+        little_bits = reversed(tuple(grouper(little_bits, 8)))
+        little_bits = tuple(chain(*little_bits))
 
+        returnList = list()
+        for i in range(0,self.size * 8):
```

Comment:
```suggestion
        for i in range(self.size * 8):
```

But overall this could be:

```python
returnList = [
    little + big
    for little, big in zip(little_bits, big_bits)
]
```

Reply:
@ebroecker, using indexes or `range()` when they are not needed is a bit of a pet peeve of mine.

Reply:
.... I can't hide my c language experiences....  
I'm learning every day more thinking 'pythonic' working with you :-) 
So thank you very much for your hints